instaclone/src/components/Authentication/withAuthenticate.js

```diff
@@ -1,9 +1,29 @@
 import React from "react";
 
-const withAuthenticate = App =>
+const withAuthenticate = App => LoginPage =>
   class extends React.Component {
+    state = {
+      loggedIn: false
+    };
+
+    componentDidMount() {
+      if (localStorage.getItem("username")) {
+        this.setState({
+          loggedIn: true
+        });
+      } else {
+        this.setState({
+          loggedIn: false
+        });
+      }
+    }
+
     render() {
-      return <App />;
+      if (this.state.loggedIn) {
+        return <App />;
+      } else {
+        return <LoginPage />;
+      }
```

Comment:
To make this a bit cleaner you could also use a [Ternary Operator](https://developer.mozilla.org/enUS/docs/Web/JavaScript/Reference/Operators/Conditional_Operator) here. To do that it would like something like this `this.state.loggedIn ? <App/> : <LoginPage />`

Reply:
thanks, will do! @Fractured2K